### DISCUSSION TITLE
Frontend/dashboard brainstorming thread

### DISCUSSION THREAD

**Discussion by brianjag** [mini-services/miniurl#113] [General]
The purpose of this discussion is to start a conversation about creating a frontend for miniurl as well as Mini Services as a project. 

The general idea is to have a marketing website for each microservice as part of a larger "toolkit".
Once we have already built several services, the main UI will be the dashboard that will display information and allow controlling the different services.

Ideally, each microservice website should include (as mentioned in #23):
- :rocket: A landing page - should be attractive and stylish, with a meaningful tagline and a brief sales pitch on what the project does, e.g. [monday.com](https://monday.com/).
- :gear: A demo of the service's core functionality. Think [Bitly](https://bitly.com/) or [Netlify Drop](https://app.netlify.com/drop).
- :books: Docs - will be divided into two subsections:
  - A usage guide - how to use and deploy the service. Check out [VuePress Documentation](https://vuepress.vuejs.org/guide/) for inspiration.
  - API - optimally, should be auto-generated from the code and hosted on a discreet repo for easy maintenance. Needs more research on how to do both (suggestions are welcome).
- :family: List of all other microservices with search and links to their websites ("part of the Mini Services project").
- :information_source: About, contributors, link to GitHub, etc.

Next steps:
1. Pick the frontend framework we would like to work with - preferably using TypeScript. Feel free to cast your vote here. **(Click on the option you want)**
[![](https://api.gh-polls.com/poll/01F0K9R85HZJ7EV5RBQBGR34P6/React)](https://api.gh-polls.com/poll/01F0K9R85HZJ7EV5RBQBGR34P6/React/vote)
[![](https://api.gh-polls.com/poll/01F0K9R85HZJ7EV5RBQBGR34P6/Vue)](https://api.gh-polls.com/poll/01F0K9R85HZJ7EV5RBQBGR34P6/Vue/vote)
[![](https://api.gh-polls.com/poll/01F0K9R85HZJ7EV5RBQBGR34P6/Angular)](https://api.gh-polls.com/poll/01F0K9R85HZJ7EV5RBQBGR34P6/Angular/vote)
[![](https://api.gh-polls.com/poll/01F0K9R85HZJ7EV5RBQBGR34P6/Svelte)](https://api.gh-polls.com/poll/01F0K9R85HZJ7EV5RBQBGR34P6/Svelte/vote)
[![](https://api.gh-polls.com/poll/01F0K9R85HZJ7EV5RBQBGR34P6/Ember)](https://api.gh-polls.com/poll/01F0K9R85HZJ7EV5RBQBGR34P6/Ember/vote)
[![](https://api.gh-polls.com/poll/01F0K9R85HZJ7EV5RBQBGR34P6/Other)](https://api.gh-polls.com/poll/01F0K9R85HZJ7EV5RBQBGR34P6/Other/vote)

2. Pick the logo - has been discussed in #43, we should create a poll as well once we have a few options to choose from.
3. Design the website tech-wise and product-wise.
4. Get some basic+ designs for it - #44
5. Start working. :smiley: 
6. Possibly get a technical writer for the docs/API sections

Further down the line: purchase domains (#31) and decide on hosting/cloud providers. We need to account for scalability by the time we have deployed the dashboard due to its complexity.

**Comment by SnirShechter** [comment]
Sounds good to me!

You can also start the frontend framework pick here (can we make polls here?) or somewhere else.

Regarding the next steps:
* I'll open the logo poll independently once I'm done with sorting out the choices - we have a designer working on it.
* Visual designs - we need to do some research and then maybe get help from the designer helping us. I can also find someone on Fiverr when the time is right.
* Let me know when an initial version of the website is done and I'll buy a domain (you can suggest one that's available and doesn't cost like my yearly salary).

**P.S** Github Discussions has 0 exposure. Can you post a link to this discussion on our Slack and on Facebook (Pull Request community, Open Source Israel community and anywhere else we want)? I can post it myself if you'd like.

**Reply by brianjag** [reply]
Great! I was thinking about using the [GitHub Polls App](https://github.com/apps/polls) to have it better organized and traceable, and to allow more people to participate, but I would need permissions to install it on the organization. You can add it yourself if you want, it's just a couple of clicks. Afterwards I will create the poll in #23.

BTW, the Jamstack guide maintains a [list of the most popular SSGs](https://jamstack.org/generators/) and I believe any one of the top 5 mentioned there would be perfect for our needs.

About the domain, I'm very happy with [Namecheap](https://www.namecheap.com/) for my own personal website (although I've heard good things about [Porkbun](https://porkbun.com/) and they seem to have better promotions), but the price will ultimately depend on the TLD. I personally prefer `.org` to reflect the open source nature of the project. `.xyz` is also a nice option and [it is available at a very good price](https://porkbun.com/checkout/search?q=mini-services.xyz&tlds=&idnLanguage=&csrf_pb=cee89942ee3eaf24d9c60dadde83988e).

I will also take some time later today to post on social media.

**Reply by SnirShechter** [reply]
I've added the Polls app.

You've also already given access to the repository. You just need to approve it.

![IMG_20210303_215121.jpg](https://user-images.githubusercontent.com/26040325/109863817-c1849000-7c6a-11eb-9d61-87070e8c9c6e.jpg)

The SSG pick depends on the framework chosen. It'll probably be Docusaurus + Gastby for React or VuePress + Gridsome for Vue.

Personally, I'm not fond of `.org`. It's pretty old-looking. Maybe `.io` or `.dev`? Anyway, this can pretty simply be a poll too 🙂

**Reply by shay123g** [reply]
After reading all the correspondence I got the impression the Mini services related one to other in some manner: "...that will showcase the interactions between the different services."
I dont think this is accurate because as i see it, the mini's should be independent.
true, mini's can be used together in the sampe program and may be related, but in the core idea, the mini's should be totally independent in functionality

**Reply by Danielg212** [reply]
@SnirShechter where is the poll?

**Comment by SnirShechter** [comment]
@shay123g 

You're right, it's not a good description since they are all independent. The dashboard will probably have a dedicated page or more for each mini. The being said, the dashboard might (assuming we have enough information) cross-reference data and display it, e.g same IP using multiple minis or a single request trace that goes through both the auth service and miniUrl.

@binarica 
Can you edit it? To something similar to: "the main UI will be the dashboard that will display information and allow controlling the different services."

**Comment by Man-RP** [comment]
When does the work start?
